src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -1186,27 +1186,35 @@ public override void VisitRecursivePattern(IRecursivePatternOperation operation)
 
             foreach (var subpat in operation.PropertySubpatterns)
             {
-                var (symbol, pattern) = subpat;
-                switch (symbol)
-                {
-                    case null: // error case
-                        break;
-                    case IFieldSymbol field:
-                    case IPropertySymbol prop:
-                    case IErrorTypeSymbol error:
-                        break;
-                    default:
-                        Assert.True(false, $"Unexpected symbol {symbol}");
-                        break;
-                }
+                Assert.True(subpat is IPropertySubpatternOperation);
             }
 
             IEnumerable<IOperation> children = operation.DeconstructionSubpatterns.Cast<IOperation>();
-            children = children.Concat(operation.PropertySubpatterns.Select(x => x.Item2));
+            children = children.Concat(operation.PropertySubpatterns);
 
             AssertEx.Equal(children, operation.Children);
         }
 
+        internal override void VisitPropertySubpattern(IPropertySubpatternOperation operation)
+        {
+            Assert.True(operation.Member is IMemberReferenceOperation || operation.Member is IInvalidOperation);
+            var member = (IMemberReferenceOperation)operation.Member;
```

Comment:
>(IMemberReferenceOperation)operation.Member; [](start = 25, length = 44)

Is this going to throw when ```operation.Member is IInvalidOperation```?

Reply:
Is this not getting caught by tests? Even when IOperation hook is enabled?

---
In reply to: [254480084](https://github.com/dotnet/roslyn/pull/33178#discussion_r254480084) [](ancestors = 254480084)

Reply:
I should probably run these tests through the hook, I haven't. But no, this visitor isn't called on our regular IOperation tests.

---
In reply to: [254480510](https://github.com/dotnet/roslyn/pull/33178#discussion_r254480510) [](ancestors = 254480510,254480084)